application/common/models/Mfa.php

```diff
@@ -334,10 +341,18 @@ public static function create(int $userId, string $type, string $label = null, s
             throw new BadRequestHttpException('Manager email must be valid for this MFA type');
         }
 
+
+        if ($type == self::TYPE_RECOVERY) {
+            $validator = new EmailValidator();
+            if (!$validator->validate($recoveryEmail)) {
+                throw new BadRequestHttpException('Recovery email must be valid for this MFA type.', 1742328138);
+            }
+        }
+
         $existing = self::findOne(['user_id' => $userId, 'type' => $type, 'verified' => 1]);
 
         if ($existing instanceof Mfa) {
-            if ($type == self::TYPE_BACKUPCODE || $type == self::TYPE_MANAGER || $type == self::TYPE_WEBAUTHN) {
+            if ($type == self::TYPE_BACKUPCODE || $type == self::TYPE_MANAGER || $type == self::TYPE_RECOVERY || $type == self::TYPE_WEBAUTHN) {
                 $mfa = $existing;
             } else {
                 throw new ConflictHttpException('An MFA of type ' . $type . ' already exists.', 1551190694);
```

Comment:
It's probably time to reverse this logic
```suggestion
            if ($type == self::TYPE_TOTP) {
                throw new ConflictHttpException('An MFA of type ' . self::TYPE_TOTP . ' already exists.', 1551190694);
            } else {
                $mfa = $existing;
```

Reply:
You could either do that or extract a method named for what question this is trying to ask. For example, `isTypeThatUsesChildRecords($type)` or something like that. I'm not sure if that's the exact question this if-condition is trying to ask, but I think it's something close to that. Feel free to adjust wording as you see fit if you choose to go this route.

I slightly prefer a whitelisted approach (= it's one of these) rather than blacklisted (it's not TOTP), but I also respect @briskt's input on this. It has certainly gotten too long for a single if-condition at this point.

Reply:
True, we could easily use a switch here.

Reply:
A switch isn't what either of us recommended. It would only make a very long function longer.